### PR TITLE
PR: Add precommits with black and import sort

### PR DESCRIPTION
**Pull request by goanpeca**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Contributing
```
Comment (goanpeca) [inline CONTRIBUTING.md:1]: @runfalk should I add the dev environment settings?

I usually use conda for everytthing so if you use pipenv or venv I could use a hand here, I am rusty on that side :-p

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+We use precommit hooks to ensure code style and format.
+
+* Install precommit from pip
+
+  ```
+  pip install pre-commit
+  ```
+
+* Now after each commit, the style hooks will run and auto format the code.
```
Comment (nixjdm) [inline CONTRIBUTING.md:13]: We also need to have the hooks installed (and missing a dash). We could also mention how to run the hooks without an actual commit:

----------

## Pre-commit

Install pre-commit from pip:

  ```bash
  pip install pre-commit
  pre-commit install
  ```

 Now after each commit, the style hooks will run and auto format the code.

  You can also manually run the pre-commit hooks without a commit with `pre-commit run -a`

File: CONTRIBUTING.md (hunk 1 above)
Comment (nixjdm) [reply CONTRIBUTING.md:1]: It all should be the same with the assumption that they have a compatible python and pip installed. I don't think we need to give much guidance on conda/venv/pipenv/poetry/...

Comment (goanpeca) [reply CONTRIBUTING.md:1]: Fair enough

**Comment by runfalk** [comment]
I think this looks good :+1:

**Pull request merged**